### ISSUE TITLE
Feature Request: Remote debugging access to the containers

### ISSUE THREAD

**Issue by denniszielke** [microsoft/azure-container-apps#632]
**Is your feature request related to a problem? Please describe**.  
Some of the complex microservices scenarios cannot be simulated locally but debugging is still an important capability to understand what is happening in a distributed system.
Same as AppServices/ Azure Spring Apps we are asking for remote debugging functionality to be available in Azure Container Apps too.

**Describe the solution you'd like**.    
We would like to see an option to enable debugging (default turned off) on the Environment or Container App Level - assuming the user has remote debugging permissions (which should be an assignable action via Azure RBAC)

**Describe alternatives you've considered**.   
Remote shell is helpful but sometimes a debugger is the only way.

**Comment by tmcgannon** [comment]
@SophCarp: Before Azure Container Applications, I was using AKS. Using the Kubernetes vscode extension I was able to attach to my running process and debug my code live via tunneling (https://cloudblogs.microsoft.com/opensource/2021/06/02/enabling-local-tunnel-debugging-with-the-kubernetes-extension-for-vs-code/). This was extremely valuable. Is debug via tunneling possible with Azure Container Applications?

**Comment by SophCarp** [comment]
@tmcgannon thanks for the info! We are not planning on supporting tunneling. We do have an item in our backlog to make some forms of remote debugging possible through port forwarding. We don't have any immediate plans that I'm aware of. 

@anthonychu for visibility

**Comment by RicardoNiepel** [comment]
@SophCarp @tmcgannon @anthonychu Do you have any update on this whitespace?

**Comment by tmcgannon** [comment]
> @SophCarp @tmcgannon @anthonychu Do you have any update or workaround on this whitespace?

I have no workaround but very interested in finding one.

**Comment by EinmalIM** [comment]
Yes, remote debugging would be really helpful or at least creating a memory dump to analyze state and callstacks of an instance. In this area App Services are much easier to analyze.

Any update on plans to make remote debugging possible?

**Comment by electricbolt** [comment]
+1 for remote debugging a java app deployed as an Azure Container app thanks.
